### PR TITLE
system variable: add tidb_opt_skew_distinct_agg

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)
system variable: add tidb_opt_skew_distinct_agg

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.2 (TiDB 6.2 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/10564 https://github.com/pingcap/docs-cn/pull/10797
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -1772,6 +1772,13 @@ explain select * from t where age=5;
 3 rows in set (0.00 sec)
 ```
 
+### tidb_opt_skew_distinct_agg <span class="version-mark">New in v6.2.0</span>
+
+- Scope: SESSION | GLOBAL
+- Persists to cluster: No
+- Default valiue: `OFF`
```
Comment (Oreoxmt) [inline system-variables.md:1779]:
```suggestion
- Type: Boolean
- Default value: `OFF`
```

File: system-variables.md [hunk 2]
```diff
@@ -1772,6 +1772,13 @@ explain select * from t where age=5;
...
+- This variable sets whether the optimizer rewrites the aggregate functions with `DISTINCT` to the two-level aggregate functions, such as rewriting `SELECT b, count(DISTINCT a) FROM t GROUP BY b` to `SELECT b, count(a) FROM (SELECT b, a FROM t GROUP BY b, a) t GROUP BY b`. When the aggregation coloumn has serious skew and the `DISTINCT` column has many different values, this rewriting can avoid the data skew in the query execution process and improve the query performance. This optimization is effective **only for TiFlash**.
```
Comment (Oreoxmt) [inline system-variables.md:1780]:
```suggestion
- This variable sets whether the optimizer rewrites the aggregate functions with `DISTINCT` to the two-level aggregate functions, such as rewriting `SELECT b, COUNT(DISTINCT a) FROM t GROUP BY b` to `SELECT b, COUNT(a) FROM (SELECT b, a FROM t GROUP BY b, a) t GROUP BY b`. When the aggregation coloumn has serious skew and the `DISTINCT` column has many different values, this rewriting can avoid the data skew in the query execution and improve the query performance. This optimization is effective **only for TiFlash**.
```

**Review by Oreoxmt — APPROVED** [review]
LGTM

File: system-variables.md [hunk 3]
```diff
@@ -1768,6 +1768,14 @@ explain select * from t where age=5;
 3 rows in set (0.00 sec)
 ```
 
+### tidb_opt_skew_distinct_agg <span class="version-mark">New in v6.2.0</span>
+
+- Scope: SESSION | GLOBAL
+- Persists to cluster: No
```
Comment (fixdb) [inline system-variables.md:1774]: Should be YES. The Chinese version needs to be updated too.

Comment (TomShawn) [reply system-variables.md:1774]:
```suggestion
- Persists to cluster: Yes
```

**Review by fixdb — APPROVED** [review]
+1

File: system-variables.md [hunk 4]
```diff
@@ -1768,6 +1768,14 @@ explain select * from t where age=5;
 3 rows in set (0.00 sec)
 ```
 
+### tidb_opt_skew_distinct_agg <span class="version-mark">New in v6.2.0</span>
+
+- Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
+- Type: Boolean
+- Default value: `OFF`
+- This variable sets whether the optimizer rewrites the aggregate functions with `DISTINCT` to the two-level aggregate functions, such as rewriting `SELECT b, COUNT(DISTINCT a) FROM t GROUP BY b` to `SELECT b, COUNT(a) FROM (SELECT b, a FROM t GROUP BY b, a) t GROUP BY b`. When the aggregation coloumn has serious skew and the `DISTINCT` column has many different values, this rewriting can avoid the data skew in the query execution and improve the query performance. This optimization is effective **only for TiFlash**.
```
Comment (shichun-0415) [inline system-variables.md:1777]:
```suggestion
- This variable sets whether the optimizer rewrites the aggregate functions with `DISTINCT` to the two-level aggregate functions, such as rewriting `SELECT b, COUNT(DISTINCT a) FROM t GROUP BY b` to `SELECT b, COUNT(a) FROM (SELECT b, a FROM t GROUP BY b, a) t GROUP BY b`. When the aggregation column has serious skew and the `DISTINCT` column has many different values, this rewriting can avoid the data skew in the query execution and improve the query performance. This optimization is effective **only for TiFlash**.
```

File: system-variables.md [hunk 5]
```diff
@@ -1768,6 +1768,14 @@ explain select * from t where age=5;
 3 rows in set (0.00 sec)
 ```
 
+### tidb_opt_skew_distinct_agg <span class="version-mark">New in v6.2.0</span>
+
+- Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
+- Type: Boolean
+- Default value: `OFF`
+- This variable sets whether the optimizer rewrites the aggregate functions with `DISTINCT` to the two-level aggregate functions, such as rewriting `SELECT b, COUNT(DISTINCT a) FROM t GROUP BY b` to `SELECT b, COUNT(a) FROM (SELECT b, a FROM t GROUP BY b, a) t GROUP BY b`. When the aggregation column has serious skew and the `DISTINCT` column has many different values, this rewriting can avoid the data skew in the query execution and improve the query performance. This optimization is effective **only for TiFlash**.
```
Comment (TomShawn) [inline system-variables.md:1777]:
```suggestion
- This variable sets whether the optimizer rewrites the aggregate functions with `DISTINCT` to the two-level aggregate functions, such as rewriting `SELECT b, COUNT(DISTINCT a) FROM t GROUP BY b` to `SELECT b, COUNT(a) FROM (SELECT b, a FROM t GROUP BY b, a) t GROUP BY b`. When the aggregation column has serious skew and the `DISTINCT` column has many different values, this rewriting can avoid the data skew in the query execution and improve the query performance. 
```

File: system-variables.md [hunk 6]
```diff
@@ -1768,6 +1768,14 @@ explain select * from t where age=5;
 3 rows in set (0.00 sec)
 ```
 
+### tidb_opt_skew_distinct_agg <span class="version-mark">New in v6.2.0</span>
+
```
Comment (TomShawn) [inline system-variables.md:1772]:
```suggestion

> **Note:**
>
> The optimization enabled by this variable is effective **only for TiFlash**.

```

File: system-variables.md [hunk 7]
```diff
@@ -1768,6 +1768,18 @@ explain select * from t where age=5;
...
+> **Note:**
+>
+> The optimization enabled by this variable is effective **only for TiFlash**.
```
Comment (TomShawn) [inline system-variables.md:1775]:
```suggestion
> The query performance optimization by enabling this variable is effective **only for TiFlash**.
```

**Pull request merged**